pex/interpreter_constraints.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# A library of functions for filtering Python interpreters based on compatibility constraints
+
+from .interpreter import PythonIdentity
+
+
+def _matches(interpreter, filters, meet_all_constraints=False):
```

Comment:
I added meet_all_constraints because for some reason the Pants usage of this code (currently in Pants codebase) only requires a match against one of the filters to be considered a "matching interpreter".